src/Adapter/Order/OrderDetailUpdater.php

```diff
@@ -0,0 +1,471 @@
+    }
+
+    /**
+     * @param OrderDetail $orderDetail
+     * @param Number $priceTaxExcluded
+     * @param Number $priceTaxIncluded
+     * @param int $roundType
+     * @param int $computingPrecision
+     *
+     * @throws OrderException
+     */
+    private function applyPriceUpdate(
+        OrderDetail $orderDetail,
+        Number $priceTaxExcluded,
+        Number $priceTaxIncluded,
+        int $roundType,
+        int $computingPrecision
+    ): void {
+        $floatPriceTaxExcluded = (float) (string) $priceTaxExcluded;
+        $floatPriceTaxIncluded = (float) (string) $priceTaxIncluded;
+
+        $orderDetail->product_price = $orderDetail->unit_price_tax_excl = $floatPriceTaxExcluded;
+        $orderDetail->unit_price_tax_incl = $floatPriceTaxIncluded;
+        switch ($roundType) {
+            case Order::ROUND_TOTAL:
+                $orderDetail->total_price_tax_excl = $floatPriceTaxExcluded * $orderDetail->product_quantity;
+                $orderDetail->total_price_tax_incl = $floatPriceTaxIncluded * $orderDetail->product_quantity;
+
+                break;
+            case Order::ROUND_LINE:
+                $orderDetail->total_price_tax_excl = Tools::ps_round($floatPriceTaxExcluded * $orderDetail->product_quantity, $computingPrecision);
+                $orderDetail->total_price_tax_incl = Tools::ps_round($floatPriceTaxIncluded * $orderDetail->product_quantity, $computingPrecision);
+
+                break;
+
+            case Order::ROUND_ITEM:
+            default:
+                $orderDetail->product_price = $orderDetail->unit_price_tax_excl = Tools::ps_round($floatPriceTaxExcluded, $computingPrecision);
+                $orderDetail->unit_price_tax_incl = Tools::ps_round($floatPriceTaxIncluded, $computingPrecision);
+                $orderDetail->total_price_tax_excl = $orderDetail->unit_price_tax_excl * $orderDetail->product_quantity;
+                $orderDetail->total_price_tax_incl = $orderDetail->unit_price_tax_incl * $orderDetail->product_quantity;
+
+                break;
+        }
+
+        if (!$orderDetail->update()) {
+            throw new OrderException('An error occurred while editing the product line.');
+        }
+    }
+
+    /**
+     * @param Order $order
+     * @param int $productId
+     * @param int $combinationId
+     * @param Number $priceTaxExcluded
+     * @param Number $priceTaxIncluded
+     * @param int $roundType
+     * @param int $computingPrecision
+     * @param Address $taxAddress
+     *
+     * @throws OrderException
+     */
+    private function applyIdenticalUpdates(
+        Order $order,
+        int $productId,
+        int $combinationId,
+        Number $priceTaxExcluded,
+        Number $priceTaxIncluded,
+        int $roundType,
+        int $computingPrecision,
+        Address $taxAddress
+    ): void {
+        $identicalOrderDetails = $this->getIdenticalOrderDetails($order, $productId, $combinationId);
+        if (empty($identicalOrderDetails)) {
```

Comment:
I think i'd go for this check inside `getIdenticalOrderDetails` and maybe `array_shift` it

Reply:
I would if I only wanted the first identical OrderDetail, but I need the whole list since then we loop through the whole `$identicalOrderDetails` array (L319)